### PR TITLE
replace raw image output with urls

### PR DESCRIPTION
**Pull request by deep-diver**
let me know what you think!

### CODE REVIEW

File: server/similarity_utils.py [hunk 1]
```diff
@@ -40,9 +40,8 @@ def perform_sim_search(self, images, query_phrase, top_k=3):
         logits_per_image = outputs.logits_per_image
 
         top_indices, top_scores = self.sort_scores(logits_per_image, top_k)
-        top_images = [images[index] for index in top_indices]
```
Comment (sayakpaul) [inline server/similarity_utils.py:43]: We don't need to compute this if we are returning URLs only.

File: server/pixabay_utils.py [hunk 2]
```diff
@@ -39,14 +40,16 @@ def fetch_images_tag(pixabay_search_keyword, num_images):
...
         response = requests.get(imageurl)
         image = Image.open(BytesIO(response.content)).convert("RGB")
         all_images.append(image)
+        all_image_urls.append(imageurl)
 
     end_time = time.time() - start_time
     logging.info(f"Fetched individual results in {end_time:.3f} " f"seconds.")
 
-    return all_images
+    return all_images, all_image_urls
```
Comment (sayakpaul) [inline server/pixabay_utils.py:56]: Why return images at all in that case?

File: server/perform_search.py [hunk 3]
```diff
@@ -15,10 +15,12 @@ def get_similar_images(self, keyword, semantic_query, pixabay_max, top_k):
         :param top_k: Top-k images to return.
         :return: List of top-k PIL images.
```
Comment (sayakpaul) [inline server/perform_search.py]: The return part and the short description need to be changed then.

File: server/perform_search.py [hunk 4]
```diff
@@ -15,10 +15,12 @@ def get_similar_images(self, keyword, semantic_query, pixabay_max, top_k):
...
         """
-        keyword_images = fetch_images_tag(keyword, pixabay_max)
+        keyword_images, keyword_image_urls = fetch_images_tag(keyword, pixabay_max)
```
Comment (sayakpaul) [inline server/perform_search.py:18]: Images could be discarded then?

**Review by sayakpaul — CHANGES_REQUESTED** [review]
@deep-diver I am okay with replacing the raw images with URLs. Please review the requested changes.

File: server/perform_search.py (hunk 4 above)
Comment (deep-diver) [reply server/perform_search.py:18]: Images still need to be there for being fed into the model?

File: server/similarity_utils.py (hunk 1 above)
Comment (deep-diver) [reply server/similarity_utils.py:43]: Sorry about this, my bad to fix things too quickly. Let me fix this tomorrow

File: server/perform_search.py (hunk 4 above)
Comment (sayakpaul) [reply server/perform_search.py:18]: Correct. My bad.

File: server/pixabay_utils.py (hunk 2 above)
Comment (deep-diver) [reply server/pixabay_utils.py:56]: This is the same reason that we have to feed images into the model, so we need both raw image data and the urls

File: server/similarity_utils.py (hunk 1 above)
Comment (deep-diver) [reply server/similarity_utils.py:43]: @sayakpaul 
do you mean we don't need `top_indices, top_scores = self.sort_scores(logits_per_image, top_k)`?
I guess we still need `top_indices` to refer which URLs to return, 
so I just removed `top_images = [images[index] for index in top_indices]` part.

what do you think?

File: server/perform_search.py (hunk 3 above)
Comment (deep-diver) [reply server/perform_search.py]: sure! I have just commited this change, so please check it out!

File: server/similarity_utils.py (hunk 1 above)
Comment (sayakpaul) [reply server/similarity_utils.py:43]: Yes, got it.

**Review by sayakpaul — APPROVED** [review]

**Pull request merged**